CRM/Financial/BAO/Order.php

```diff
@@ -231,9 +233,13 @@ protected function calculateLineItems(): array {
     }
 
     foreach ($this->getPriceOptions() as $fieldID => $valueID) {
+      if (!isset($this->priceSetID)) {
+        $this->setPriceSetID(PriceField::get()->addSelect('price_set_id')->addWhere('id', '=', $fieldID)->execute()->first()['price_set_id']);
+      }
       $throwAwayArray = [];
       // @todo - still using getLine for now but better to bring it to this class & do a better job.
-      $lineItems[$valueID] = CRM_Price_BAO_PriceSet::getLine($params, $throwAwayArray, $this->getPriceSetID(), $this->getPriceFieldSpec($fieldID), $fieldID, 0)[1][$valueID];
+      $newLine = reset(CRM_Price_BAO_PriceSet::getLine($params, $throwAwayArray, $this->getPriceSetID(), $this->getPriceFieldSpec($fieldID), $fieldID, 0)[1]);
+      $lineItems[$newLine['price_field_value_id']] = $newLine;
```

Comment:
This turns out to be a bug on is_enter_qty line items - valueId is the qty not the id in those cases. Currently only accessed from MembershipRenewal form